### ISSUE TITLE
prefix-in renaming fails when replacement token is empty [bug, feature request]

### ISSUE THREAD

**Issue by mfelleisen** [racket/drracket#137]
Here is the result of renaming the prefix "abc:" to "", the empty string: 
```
#lang racket
(require (prefix-in || "foo.rkt"))
(+ ||xyz (||xyz 10))
```

I expect to get this: 
```#lang racket
(require "foo.rkt")
(+ xyz (xyz 10))```

When you run search-and-replace in drracket with "abc:" to "" it does that (minus the prefix-in removal). 

So it's possible that this is not a bug report but a feature request for "remove prefix-in".

**Comment by rfindler** [comment]
Renaming currently assumes that the identifier being renamed subscribes to the racket symbol syntax. Probably that's a bad idea and we should just take the literal text that is typed in and replace it with that at each point.

So: if you wanted to rename this program:

```
#lang racket
(define (f x) x)
```

to this one:

```
#lang racket
(define (f x\ x) x\ x)
```

You would have to type `x\ x` into the dialog, instead of typing `x x` (which is what you would do currently).

If we made that change, then the renaming you did would produce this program:

```
#lang racket
(require (prefix-in "foo.rkt"))
(+ xyz (xyz 10))
```

and then you'd get a syntax error that would take right to the `prefix-in`.

How does that sound?

**Comment by mfelleisen** [comment]
> On Aug 17, 2017, at 10:49 AM, Robby Findler <notifications@github.com> wrote:
> 
> Renaming currently assumes that the identifier being renamed subscribes to the racket symbol syntax. Probably that's a bad idea and we should just take the literal text that is typed in and replace it with that at each point.
> 
> So: if you wanted to rename this program:
> 
> #lang racket
> (define (f x) x)
> 
> to this one:
> 
> #lang racket
> (define (f x\ x) x\ x)
> 
> You would have to type x\ x into the dialog, instead of typing x x (which is what you would do currently).
> 
> If we made that change, then the renaming you did would produce this program:
> 
> #lang racket
> (require (prefix-in "foo.rkt"))
> (+ xyz (xyz 10))
> 
> and then you'd get a syntax error that would take right to the prefix-in.
> 
> How does that sound?


My preference is ‘remove prefix’ but this is a close second.

**Comment by dfeltey** [comment]
I think we could probably add a remove-prefix operation. More generally, I
think there are probably a bunch of refactoring-like operations that make
sense for require forms, but I'm not sure exactly what the interface for
using them should be, continually adding right-click menu-items seems like
a bad idea.

**Comment by rfindler** [comment]
I agree with @dfeltey but went ahead with the change suggested above anyway because it doesn't change the set of things in the popup menu and it both helps with the immediate problem (of removing a `prefix-in`) and with making the renaming operation more language agnostic.

**Comment by mfelleisen** [comment]
I’d he happy to click! 

I have a lot of suggestions for refactoring transformations and some are Racket-specific. With Frank Tip around now, I think I may even know how to publish papers on that :-)
